### PR TITLE
Remove nwbinspector dependency

### LINKED ISSUE #651: Install neuroconv without dandi
Right now dandi is part of our basic dependencies and naturally it comes itself bundled with a bunch of libraries for handling downloading and uploading. 

For the use case of just transforming data within an analysis pipeline none of those dependencies are really needed. 

Maybe it would be worth to add an option to install neuroconv just as a t**ransformation tool** that other libraries can use without the dandi / web / tools-for-uploading-and-downloading data dependencies.

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #651

### CODE REVIEW

File: src/neuroconv/tools/data_transfers/_globus.py [hunk 1]
```diff
@@ -22,7 +20,7 @@ def get_globus_dataset_content_sizes(
 
     Returns dictionary whose keys are file names and values are sizes in bytes.
     """
-    assert HAVE_GLOBUS, "You must install the globus CLI (pip install globus-cli)!"
+    assert is_package_installed(module_name="globus_cli"), "You must install the globus CLI (pip install globus-cli)!"
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/data_transfers/_globus.py:23]: Adjusted wording from `module` (as in `nwbinspector`) to `package` (as in `neuroconv`), which is also more accurate TBH

```suggestion
    assert is_package_installed(package_name="globus_cli"), "You must install the globus CLI (pip install globus-cli)!"
```

File: src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 )
 _available_hdf5_filters = set(_base_hdf5_filters - _excluded_hdf5_filters)
 AVAILABLE_HDF5_COMPRESSION_METHODS = {filter_name: filter_name for filter_name in _available_hdf5_filters}
-if is_module_installed(module_name="hdf5plugin"):
+if is_package_installed(module_name="hdf5plugin"):
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py:21]:
```suggestion
if is_package_installed(package_name="hdf5plugin"):
```

File: src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ class Config:
     )
 
     def get_data_io_kwargs(self) -> Dict[str, Any]:
-        if is_module_installed(module_name="hdf5plugin"):
+        if is_package_installed(module_name="hdf5plugin"):
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py:58]:
```suggestion
        if is_package_installed(package_name="hdf5plugin"):
```

File: tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_helpers/test_get_default_dataset_io_configurations.py [hunk 4]
```diff
@@ -234,7 +234,8 @@ def test_configuration_on_compass_direction(iterator: callable, backend: Literal
 
 
 @pytest.mark.skipif(
-    not is_module_installed(module_name="ndx_events"), reason="The extra testing package 'ndx-events' is not installed!"
+    not is_package_installed(module_name="ndx_events"),
```
Comment (CodyCBakerPhD) [inline tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_helpers/test_get_default_dataset_io_configurations.py:237]:
```suggestion
    not is_package_installed(package_name="ndx_events"),
```

**Review by h-mayorquin — APPROVED** [review]
LGTM

**Pull request merged**